jupyter_server/auth/authorizer.py

```diff
@@ -39,9 +39,10 @@ def is_authorized(self, handler: JupyterHandler, user: str, action: str, resourc
         Parameters
         ----------
         user : usually a dict or string
-            A truthy model representing the authenticated user.
+            A truthy object representing the authenticated user,
+            as returned by :meth:`.IdentityProvider.get_user`.
```

Comment:
The type in this signature for the `user` parameter should be `Any`.  I wonder if there should be a method on IdentityProvider called `get_username(user: Any) -> str:` that Authorizers (and others) can use to access the definitive `username` value?  Otherwise, it seems like Authorizers and IdentityProviders need to be co-implemented so long as `IdentityProvider.get_user()` returns an `Any`.

Reply:
We have `get_identity` (whatever we call the filled version) which transforms the opaque Any into a strict model, so username can be accessed with `get_identity(user)['username']`.

> Otherwise, it seems like Authorizers and IdentityProviders need to be co-implemented so long as IdentityProvider.get_user() returns an Any.

I think that's going to be true in practice for almost any Authorizer that takes users into account (i.e. except for "everybody read-only" Authorizers that ignore the identity).

Reply:
This now returns a User dataclass, which defines the fields in the model. The backward-compat lies in the wrapper of LoginHandler.get_user, which may return a simple string, which will be interpreted as username.